automation_testing/pages/AdvancedFiltersComponent.ts

```diff
@@ -0,0 +1,225 @@
+		return await this.comparisonCategoriesOptions.locator("li").count();
+	}
+
+
+	async getSpecificMenuItemForSingleSelectCategoryOption(menuPosition: number): Promise<string> {
+		return this.menuOptions.locator("li").nth(menuPosition).textContent();
+	}
+
+	async getAllCategoriesForSingleSelectCategoryOption(): Promise<string[]> {
+		const numberOfOptions = await this.getNumberOfSingleSelectCategoryOptions();
+		const options = [];
+		for (let i = 0; i < numberOfOptions; i++) {
+			options.push(await this.getSpecificMenuItemForSingleSelectCategoryOption(i));
+		}
+		return options;
+	}
+
+	async selectARandomCategoryForSingleSelectCategoryOption(): Promise<string> {
+		const positionCategorySelected = await randomIntFromInterval(1, await this.getNumberOfSingleSelectCategoryOptions());
+		const categorySelected = await this.getSpecificMenuItemForSingleSelectCategoryOption(positionCategorySelected);
+		await this.menuOptions.locator("li").nth(positionCategorySelected).click();
+		return categorySelected;
+	}
+
+	/*async selectAllAdvancedFilters(): Promise<void> {
+		await this.singleSelectCategoryOption.click();
+		await this.categoryWithComparisonOption.click();
+		await this.titleOption.click();
+		await this.createdOption.click();
+		await this.updatedOption.click();
+		await this.titleWithComparisonOption.click();
+	}*/
+
+	async selectAllAdvancedFilters(): Promise<void> {
+		const filters = await this.checkboxOptions.elementHandles();
+		for (const filter of filters) {
+			if ((await (await filter.$(this.checkboxLocator)).isChecked()) == false) {
+				await (await filter.$(this.checkboxLocator)).check();
+			}
+		}
+	}
+
+	async getSelectedValueForSingleSelectCategoryOption(): Promise<string> {
+		return (await this.optionalFilters.textContent()).split(":")[1];
+	}
+
+	async getAllSelectedValuesForAdvancedFilters(): Promise<string[]> {
+		const filters = await this.optionalFilters.elementHandles();
+		const result = [];
+		for (const filter of filters) {
+			result.push((await (await filter.$(this.advancedFilterLocator)).textContent()).split(":")[1]);
+		}
+		return result;
+	}
+
+	async selectFirstCategoriesForCategoryWithComparisonOption(): Promise<string> {
+		const selectedCategory = this.comparisonCategoriesOptions.nth(0).textContent();
+		await this.comparisonCategoriesOptions.nth(0).check();
+		return selectedCategory;
+	}
+
+	async selectComparisonOption(option: string): Promise<void> {
+		await this.comparisonDropdown.click();
+		switch (option) {
+			case "In":
+				await this.dropdownOption.nth(0).click();
+				break;
+			case "Not In":
+				await this.dropdownOption.nth(1).click();
+				break;
+			case "All":
+				await this.dropdownOption.nth(2).click();
+				break;
+			case "Exists":
+				await this.dropdownOption.nth(3).click();
+				break;
+			case "Not Exists":
+				await this.dropdownOption.nth(4).click();
+				break;
+			default:
+				alert("Option does not exists.");
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 3 tabs but found 4.](https://app.codacy.com/gh/simpleviewinc/sv-mosaic/pullRequest?prid=9701980)
```suggestion
			alert("Option does not exists.");
```